Services/UI/OJS.Services.Ui.Business/Implementations/ParticipantsBusinessService.cs

```diff
@@ -151,6 +161,13 @@ public Task<int> GetParticipantLimitBetweenSubmissions(int participantId, int co
             .GetUserSubmissionTimeLimit(participantId, contestLimitBetweenSubmissions)
             .ToTask();
 
+    private static DateTime? CalculateParticipantParticipationEndTime(Contest contest)
```

Comment:
This is used for online contests from what i see. In this case, `ParticipationEndTime` is always `timeWithDuration` if the contest isn't already closed. Make sure that this is handled because otherwise, setting the `ParticipationEndTime` to `contest.EndTime` is not correct.